### DISCUSSION TITLE
Theia Render the caton problem

### DISCUSSION THREAD

**Discussion by woaixiaobo** [eclipse-theia/theia#12896] [General]
My requirement is that there are many controls that can drag, scale and rotate in the content area. Now I only put one such control, and when I scale and rotate, the control has obvious problems of stalling. The plug-in I use is vue3-moveable, but I have no problem in normal vue projects. Is this a problem with theia framework? Have you met?

**Comment by msujew** [comment]
I don't think we have encountered such problems yet. In the end, the widget area is a simple `div` that renders its contents using a calculated height and width. With stalling, you're meaning that the scaling/rotating takes a long time or simply stops rendering at all? You should try to use your browsers devtools to identify what is taking such a long time. It might be something in Theia, but I wouldn't think so.

**Comment by woaixiaobo** [comment]
What I mean is that the process of scaling and rotation is not smooth and slow, and the library moveable is used. Scaling actually changes the width and height of dom, that is, the width and height or rotation Angle of dom can be changed with the movement of the mouse after triggering scaling and rotation, because changing the width and height of dom will trigger reflux, and the triggering frequency is relatively high. So it leads to the deadlock, but ordinary vue scaffolding does not have this problem, I think is it because theia itself integrates a lot of extensions, so there are certain performance problems in page rendering?
